app/src/main/java/com/example/themealsapp/utils/UIState.kt

```diff
@@ -1,7 +1,7 @@
 package com.example.themealsapp.utils
 
 sealed class UIState<out T> {
-    object LOADING: UIState<Nothing>()
+    data class LOADING<T>(val message: T? = null): UIState<T>()
```

Comment:
why you need the generic in the loading?

Reply:
It is in order to be able to retrieve different objects using the same UIState. For instance, I am retrieving a List of Meals, in order to make the app scalable, I will need to retrieve List of Ingredients too.